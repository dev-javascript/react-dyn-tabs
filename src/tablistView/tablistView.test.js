import React from 'react';
import {unmountComponentAtNode} from 'react-dom';
import TablistView from './tablistView.js';
import renderer from 'react-test-renderer';
import Api from '../utils/api/api.js';
let container = document.createElement('div'),
  realUseContext;
const setMockUseContext = (op = {}) => {
  const instance = new Api({options: Object.assign({}, op)});
  React.useContext = jest.fn(() => instance);
};
beforeAll(() => {
  document.body.appendChild(container);
  realUseContext = React.useContext;
});
beforeEach(() => {});
afterEach(() => {
  unmountComponentAtNode(container);
  container.innerHTML = '';
  React.useContext = realUseContext;
});
afterAll(() => {
  document.body.removeChild(container);
  container = null;
});
describe('TabListView structure : ', () => {
  test('default options', () => {
    setMockUseContext();
    const tree = renderer
      .create(
        <div>
          <TablistView></TablistView>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('rtl option', () => {
    setMockUseContext({direction: 'rtl'});
    const tree = renderer
      .create(
        <div>
          <TablistView></TablistView>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('isVertical option', () => {
    setMockUseContext({isVertical: true});
    const tree = renderer
      .create(
        <div>
          <TablistView></TablistView>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('children props', () => {
    setMockUseContext();
    const tree = renderer
      .create(
        <div>
          <TablistView>
            <div className="rc-dyn-tabs-tablist-container" />
          </TablistView>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
