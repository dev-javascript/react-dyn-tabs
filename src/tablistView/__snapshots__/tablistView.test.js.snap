// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabListView structure :  children props 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-view rc-dyn-tabs-ltr all-themes"
    style={{}}
  >
    <div
      className="rc-dyn-tabs-tablist-container"
    />
  </div>
</div>
`;

exports[`TabListView structure :  default options 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-view rc-dyn-tabs-ltr all-themes"
    style={{}}
  />
</div>
`;

exports[`TabListView structure :  isVertical option 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-view rc-dyn-tabs-ltr rc-dyn-tabs-vertical all-themes"
    style={{}}
  />
</div>
`;

exports[`TabListView structure :  rtl option 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-view rc-dyn-tabs-rtl all-themes"
    style={{}}
  />
</div>
`;

exports[`TabListView structure :  style option 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-view rc-dyn-tabs-ltr all-themes"
    style={
      {
        "backgroundColor": "red",
      }
    }
  />
</div>
`;
